src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -149,13 +150,22 @@ public void Dispose()
 
         private void Reset(bool isDisposing = false)
         {
-            _workerCts?.Cancel();
+            if (_workerCts?.IsCancellationRequested == false)
+            {
+                _workerCts.Cancel();
```

Comment:
Good point. Although `IDisposable.IsDisposed` method is not provided. So I'll simply waive the conditional in both places.